resources/codegen/tests/generators/test_python.py

```diff
@@ -0,0 +1,681 @@
+        self, module_all, property_type
+    ):
+        property_name = f"{property_type}Property"
+        function_name = f"set{property_type.capitalize()}Property"
+        function = getattr(module_all.traits.namespace.AllPropertiesTrait, function_name)
+
+        assert inspect.isfunction(function)
+        assert (
+            function.__doc__
+            == f"""
+        Sets the {property_name} property.
+
+        A {property_type}-typed property.
+        """
+        )
+
+
+class Test_python_package_all_specifications:
+    def test_contains_declaration_namespaces(self, module_all):
+        assert inspect.ismodule(module_all.specifications.test)
+
+    def test_test_docstring_contains_declaration_description(self, module_all):
+        assert (
+            module_all.specifications.test.__doc__
+            == """
+Specification definitions in the 'test' namespace.
+
+Test specifications.
+"""
+        )
+
+    def test_test_module_specifications_are_suffixed_with_Specification(self, module_all):
+        assert inspect.isclass(module_all.specifications.test.TwoLocalTraitsSpecification)
+        assert inspect.isclass(module_all.specifications.test.OneExternalTraitSpecification)
+        assert inspect.isclass(module_all.specifications.test.LocalAndExternalTraitSpecification)
+
+
+class Test_python_package_all_specifications_test_TwoLocalTraitsSpecification:
+    def test_docstring_contains_description(self, module_all):
+        assert (
+            module_all.specifications.test.TwoLocalTraitsSpecification.__doc__
+            == """
+    A specification with two traits.
+    """
+        )
+
+    def test_trait_set_composes_target_trait_kIds(self, module_all):
+        expected = {
+            module_all.traits.namespace.NoPropertiesTrait.kId,
+            module_all.traits.anotherNamespace.NoPropertiesTrait.kId,
+        }
+        assert module_all.specifications.test.TwoLocalTraitsSpecification.kTraitSet == expected
+
+    def test_has_trait_getters_with_expected_docstring(self, module_all):
+
+        trait_one = module_all.traits.namespace.NoPropertiesTrait
+        trait_two = module_all.traits.anotherNamespace.NoPropertiesTrait
+
+        assert inspect.isfunction(
+            module_all.specifications.test.TwoLocalTraitsSpecification.namespaceNoPropertiesTrait
+        )
+        assert (
+            module_all.specifications.test.TwoLocalTraitsSpecification.namespaceNoPropertiesTrait.__doc__
+            == f"""
+        Returns the view for the '{trait_one.kId}' trait wrapped around
+        the data held in this instance.
+        """
+        )
+
+        assert inspect.isfunction(
+            module_all.specifications.test.TwoLocalTraitsSpecification.anotherNamespaceNoPropertiesTrait
+        )
+        assert (
+            module_all.specifications.test.TwoLocalTraitsSpecification.anotherNamespaceNoPropertiesTrait.__doc__
+            == f"""
```

Comment:
[consider] Here and elsewhere, pep8 line too long. Since it's in braces you can do
```suggestion
            module_all.specifications.test.TwoLocalTraitsSpecification
            .namespaceNoPropertiesTrait.__doc__
            == f"""
```

Reply:
`black` puts these back on what line if you try to split them. As we want to move to auto-formatting vs manual curation, happy to leave as is?

Reply:
Grr, I wonder if there's an option for black to get it wrap such things?  Would be useful when we get around to black-formatting the wider codebase.  But fine for now.